src/3rd_party_update.c

```diff
@@ -260,7 +301,32 @@ enum swupd_code third_party_update_main(int argc, char **argv)
 
 	/* update 3rd-party bundles */
 	ret_code = third_party_run_operation_multirepo(cmdline_option_repo, update_repos, SWUPD_NO, "update", steps_in_update);
+	if (ret_code) {
+		goto exit;
+	}
+
+	/* read the current template copy */
+	template_file = sys_path_join("%s/%s/%s", globals_bkp.path_prefix, SWUPD_3RD_PARTY_DIR, SWUPD_3RD_PARTY_TEMPLATE_FILE);
+	template = sys_mmap_file(template_file, &template_len);
+
+	if (!template || strcmp(template, SCRIPT_TEMPLATE) != 0) {
```

Comment:
mmap return is not guaranteed to be NULL terminated. Change to: `strncmp(template, SCRIPT_TEMPLATE, template_len);`

Reply:
With current implementation of sys_mmap_file this if will be true even on errors because template won't be NULL on errors.